src/components/visualization/map/GeoJSONMap.tsx

```diff
@@ -0,0 +1,75 @@
+import 'leaflet/dist/leaflet.css';
+import { GeoJSON, MapContainer, TileLayer } from 'react-leaflet';
+import L, { LatLngExpression, Layer } from 'leaflet';
+import Legend from './Legend';
+import { MapPos } from '@/types/visualization';
+import React from 'react';
+import ResetView from './ResetView';
+import { generateRandomColor } from '@/colors';
+import { useTranslations } from 'next-intl';
+
+const collectedLabels = new Map<string, string>();
+
+const standardPos: MapPos = {
+  latLng: [49.013_677_698_392_264, 8.404_375_426_378_891],
+  zoom: 13.5,
+};
+
+export default function GeoJSONMap({ geoJsonData }: { geoJsonData: GeoJSON.FeatureCollection }) {
+  const t = useTranslations('GeoJSONMap');
+  return (
+    <MapContainer
+      center={standardPos.latLng}
+      zoom={standardPos.zoom}
+      scrollWheelZoom={true}
+      zoomControl={false}
+      style={{ height: '100vh' }}
+    >
+      <TileLayer url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png" />
+      {
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+        geoJsonData && <GeoJSON data={geoJsonData} pointToLayer={pointToLayer} onEachFeature={onEach} />
+      }
+      <ResetView resetViewInput={{ title: 'Reset', pos: standardPos }} />
+      <Legend legendInput={{ title: t('legendTitle'), labels: collectedLabels }} />
+    </MapContainer>
+  );
+}
+
+// TODO: We need to somehow set the information which field of the properties holds the label for the legend.
```

Comment:
Yes, I agree. This solution can only be temporary. As you suggested, the best way if probably adding a configuration option for this.
Regarding colors we probably could also let the user configure it. Either way random colors make for an inconsistent user experience.
Both points can be addressed in a follow-up user story.